src/Board.cpp

```diff
@@ -94,10 +94,14 @@ void Board::remove_players() {
         std::find_if(mMissles.cbegin(), mMissles.cend(), [&player](const auto& missile) {
           return player->get_tank().get_body_rect().contains(missile.get_pos());
         });
-    if (it != mMissles.cend()) {
+    if (it == mMissles.cend()) {
+      return;
+    }
+    player->get_tank().take_damage((*it).get_damage());
```

Comment:
Adding new functionality invalidates the name of the method. We  are not only removing the players, but rather.... update the players?

Reply:
I renamed the function